tools/convert_datasets/a2d2.py

```diff
@@ -0,0 +1,355 @@
+        help='Skips restructuring directory structure')
+    parser.set_defaults(restruct=True)
+    parser.add_argument(
+        '--train-on-val',
+        dest='train_on_val',
+        action='store_true',
+        help='Use validation samples as training samples')
+    parser.set_defaults(train_on_val=False)
+    parser.add_argument(
+        '--nproc', default=1, type=int, help='Number of process')
+    parser.add_argument(
+        '--no-symlink',
+        dest='symlink',
+        action='store_false',
+        help='Use hard links instead of symbolic links')
+    parser.set_defaults(symlink=True)
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    """Program for making Audi's A2D2 dataset compatible with mmsegmentation.
+
+    NOTE: The input argument path must be the ABSOLUTE PATH to the dataset
+          - NOT the symbolically linked one (i.e. data/a2d2)
+
+    Segmentation label conversion:
+        The A2D2 labels are instance segmentations (i.e. car_1, car_2, ...),
+        while semantic segmentation requires categorical segmentations.
+
+        The function 'convert_TYPE_trainids()' converts all instance
+        segmentation to their corresponding categorical segmentation and saves
+        them as new label image files..
+
+    NOTE: The following segmentation classes are ignored (i.e. trainIds 255):
+          - Ego car:  A calibrated system should a priori know what input
+                      region corresponds to the ego vehicle.
+          - Blurred area: Ambiguous semantic.
+          - Rain dirt: Ambiguous semantic.
+
+    Directory restructuring:
+        A2D2 files are not arranged in the required 'train/val/test' directory
+        structure.
+
+        The function 'restructure_a2d2_directory' creates a new compatible
+        directory structure in the root directory, and fills it with symbolic
+        links or file copies to the input and segmentation label images.
+
+    Example usage:
+        python tools/convert_datasets/a2d2.py path/to/camera_lidar_semantic
+    """
+    args = parse_args()
+    a2d2_path = args.a2d2_path
+    out_dir = args.out_dir if args.out_dir else a2d2_path
+    mmcv.mkdir_or_exist(out_dir)
+
+    # Create a list of filepaths to all original labels
+    # NOTE: Original label files have a number before '.png'
+    label_filepaths = glob.glob(osp.join(a2d2_path, '*/label/*/*[0-9].png'))
+
+    # Convert segmentation images to the Cityscapes 'TrainIds' values
+    if args.convert:
+        if args.nproc > 1:
+            mmcv.track_parallel_progress(convert_a2d2_trainids,
+                                         label_filepaths, args.nproc)
+        else:
+            mmcv.track_progress(convert_a2d2_trainids, label_filepaths)
+
+    # Restructure directory structure into 'img_dir' and 'ann_dir'
+    if args.restruct:
+        restructure_a2d2_directory(out_dir, VAL_SPLIT, args.train_on_val,
+                                   args.symlink)
+
```

Comment:
Maybe we can add assert for train-val-test samples.

Reply:
Done
```
for r in [val_ratio, test_ratio]:
        assert r >= 0. and r < 1., f'Invalid ratio {r}'
```